tools/parse/src/main/scala/org/allenai/nlpstack/parse/poly/fsm/EmbeddedClassifier.scala

```diff
@@ -96,30 +85,58 @@ class DTCostFunctionTrainer(
       trainingVectorSource.groupVectorIteratorsByTask
 
     var progressCounter = 1
-    (taskTrainingVectors map {
-      case (task, trainingVectorIter) => {
-        val trainingVectors = trainingVectorIter.toIterable
-        println(s"Task ${progressCounter} of ${trainingVectorSource.tasks.size}" +
-          s"(${task.filenameFriendlyName}) has ${trainingVectors.size} training vectors")
+    var taskClassifierSeq: Seq[(ClassificationTask, File)] = Seq() // TODO: do we need a var here?
+    taskTrainingVectors foreach {
+      case (task, trainingVectorIter) =>
+        val classifier: TransitionClassifier =
+          trainClassifier(progressCounter, task, trainingVectorIter)
+        val tempFile: File = File.createTempFile("temp", "ecl");
+        Resource.using(new PrintWriter(tempFile)) { writer =>
+          writer.println(classifier.toJson.compactPrint)
+        }
         progressCounter += 1
-        val vectors: DTFeatureVectorSource =
-          createDTFeatureVectorSource(task, trainingVectors.iterator)
-        println("Now training.")
-        val inducedClassifier: ProbabilisticClassifier = classifierTrainer(vectors)
-        val featureMap: Seq[(Int, FeatureName)] =
-          featureNames.zipWithIndex filter {
-            case (_, featIndex) =>
-              inducedClassifier.allFeatures.contains(featIndex)
-          } map {
-            case (feat, featIndex) =>
-              (featIndex, feat)
+        println(s"Saved $tempFile.")
+        System.gc()
```

Comment:
I should probably get rid of this. I was wrangling with the garbage collector and this is an artifact of that.